### ISSUE TITLE
Incurrate Penetration Volume for GT in Obman testset

### ISSUE THREAD

**Issue by lihaoming45** [hwjiang1510/GraspTTA#6]
I admire you very much for this research work, but I encountered some problems in evaluating the Obmanc test set. 

When I evaluated the GT of Obman test, I found that the Penetration Volume calculated by my side does not correspond to your article, the value in your article is 1.70cm³, and the value calculated by my side is 2.68cm³. 

I would like to confirm the details with you, Obman test set a total of 6285 samples, each sample has Hand and Object one-to-one correspondence, I used the intersect_vox function in your evaluation code to evaluate the GT, please ask me whether my side ignored something, which led to the calculated value is different.

**Comment by hwjiang1510** [comment]
Hi, it seems your setting is correct. Have you evaluated the penetration depth of the ObMan GT?

**Comment by lihaoming45** [comment]
> Hi, it seems your setting is correct. Have you evaluated the penetration depth of the ObMan GT?
Thanks for the reply quickly!
Following your suggestion, I evaluated the penetration depth on ObMan testset GT, using the "load_batch_info" function from penetration.py.  the value is 0.0097， but I think the unit is meter, not centimeter. thus ,the calculated value by my side is 0.97cm. but the depth value in your artice is 0.01cm.
  I should report that when I running the "intersect_vox" code, the program report :"Mesh is non-watertight for contained point query!".I'm not sure whether this has an impact on the Penetration  evaluation.

**Comment by lihaoming45** [comment]
Thanks for the reply quickly!
Following your suggestion, I evaluated the penetration depth on ObMan testset GT, using the "load_batch_info" function from penetration.py. the value is 0.0097， but I think the unit is meter, not centimeter. thus ,the calculated value by my side is 0.97cm. but the depth value in your artice is 0.01cm.


I should report that when I running the "intersect_vox" code, the program report :"Mesh is non-watertight for contained point query!".I'm not sure whether this has an impact on the Penetration evaluation.


| |
lihao.ming
|
|
***@***.***
|
签名由网易邮箱大师定制
On 4/30/2022 00:37，Hanwen ***@***.***> wrote：

Hi, it seems your setting is correct. Have you evaluated the penetration depth of the ObMan GT?

**Comment by hwjiang1510** [comment]
There might be some problems with your GT. ObMan dataset doesn't provide the MANO parameters for each sample, do you directly use the hand vertices? And have you visualized your GT data? The penetration should be very small.

Non-watertight mesh is fine, I also have this warning during testing.

**Comment by lihaoming45** [comment]
Yes! I use the hand vertices directly. Thank you so much.  you mean that I should use the mano vertices output from MNAO model,right?


| |
lihao.ming
|
|
***@***.***
|
签名由网易邮箱大师定制
On 5/2/2022 06:24，Hanwen ***@***.***> wrote：

There might be some problems with your GT. ObMan dataset doesn't provide the MANO parameters for each sample, do you directly use the hand vertices? And have you visualized your GT data? The penetration should be very small.

Non-watertight mesh is fine, I also have this warning during testing.

**Comment by hwjiang1510** [comment]
Directly using vertices is okay. Have you visualized your data? penetration of GT should be small.

**Comment by lihaoming45** [comment]
Yes, I visualized the grasp data of GT, the penetration of GT is small>
I found that if I use the hand vertices directly, the calculated penetration  volume is very large (2.68cm³)， while I use the vertice output from mano model. the calculate value is small (1.46cm³)， but it still not the same value (1.70cm³) in you paper.


| |
lihao.ming
|
|
***@***.***
|
签名由网易邮箱大师定制
On 5/2/2022 13:53，Hanwen ***@***.***> wrote：

Directly using vertices is okay. Have you visualized your data? penetration of GT should be small.

**Comment by lihaoming45** [comment]
Sorry to brother again, I use the load_bat_info to evaluate the depth of Obman GT, and the calcualted value is 0.012。but i think the unit is meter right? ,which mean the value is 1.2cm. while the value in your paper is 0.01cm. 


Bu the way, I calculate the penetraion volume of Obman GT is 1.45cm³. So I think the ground truth is correct. Can you tell me what's wrong about the depth evaluation for me?


| |
lihao.ming
|
|
***@***.***
|
签名由网易邮箱大师定制
On 5/2/2022 ***@***.***> wrote：
Yes, I visualized the grasp data of GT, the penetration of GT is small>
I found that if I use the hand vertices directly, the calculated penetration  volume is very large (2.68cm³)， while I use the vertice output from mano model. the calculate value is small (1.46cm³)， but it still not the same value (1.70cm³) in you paper.


| |
lihao.ming
|
|
***@***.***
|
签名由网易邮箱大师定制
On 5/2/2022 13:53，Hanwen ***@***.***> wrote：

Directly using vertices is okay. Have you visualized your data? penetration of GT should be small.
